File: src/main/java/gestorAplicacion/comunicacion/ServicioSms.java
package gestorAplicacion.comunicacion;

/**
 * Implementación de un servicio de envío específico usando SMS.
 *
 * @author Juan Daniel Bula, Leidy Daniela Alzate, Oscar Andres Usuga
 */
public class ServicioSms extends ServicioEnvio {

    /**
     * Constructor asignando el mensaje al SMS.
     * @param mensaje mensaje a ser enviado.
     */
    public ServicioSms(String mensaje) {
        super(mensaje);
    }

    @Override
    public void enviar() {
        ServicioEnvio.mensajesEnviados.add("SMS: " + mensaje);
    }
}

File: python/src/gestorAplicacion/comunicacion/servicio_sms.py
from gestoraplicacion.comunicacion.servicio_envio import ServicioEnvio


class ServicioSms(ServicioEnvio):
    """
     Constructor asignando el mensaje al SMS.
     @param mensaje mensaje a ser enviado.
    """
    def __init__(self, mensaje):
        super().__init__(mensaje)

    def enviar(self):
        super()._mensajes_enviados.append("SMS: " + super()._mensaje)
